This patch enables connecting to unsecured WiFi networks with a .config 

Upstream-Status: Merged in master

Signed-off-by: Praneeth Bodduluri <lifeeth@gmail.com>

diff --git a/src/config.c b/src/config.c
index a4c117e..5bf5cf1 100644
--- a/src/config.c
+++ b/src/config.c
@@ -698,7 +698,16 @@ static bool load_service(GKeyFile *keyfile, const char *group,
 
 		} else
 			service->security = CONNMAN_SERVICE_SECURITY_PSK;
-	}
+	} else if (str) {
+
+		if (security != CONNMAN_SERVICE_SECURITY_NONE)
+			connman_info("Mismatch no security and "
+					"setting %s = %s",
+					SERVICE_KEY_SECURITY, str);
+
+			service->security = CONNMAN_SERVICE_SECURITY_NONE;
+	} else
+			service->security = CONNMAN_SERVICE_SECURITY_NONE;
 
 	service->config_ident = g_strdup(config->ident);
 	service->config_entry = g_strdup_printf("service_%s", service->ident);
